pyriemann_qiskit/classification.py

```diff
@@ -743,3 +750,244 @@ def predict(self, X):
         """
         labels = self._predict(X)
         return self._map_indices_to_classes(labels)
+
+
+from sklearn.base import BaseEstimator, ClassifierMixin, TransformerMixin
+
+
+class NearestConvexHull(BaseEstimator, ClassifierMixin, TransformerMixin):
+
+    """Nearest Convex Hull Classifier (NCH)
+
+    In NCH, for each class a convex hull is produced by the set of matrices
+    corresponding to each class. There is no training. Calculating a distance
+    to a hull is an optimization problem and it is calculated for each testing
+    sample (SPD matrix) and each hull/class. The minimal distance defines the
+    predicted class.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.0
+    """
+
+    def __init__(self, n_jobs=6, n_hulls=3, n_samples_per_hull=10):
```

Comment:
We should add a `optimizer=ClassicalOptimizer()` argument, used in `qdistance_logeuclid_to_convex_hull`.

Reply:
The optimizer as a global scope:

![image](https://github.com/pyRiemann/pyRiemann-qiskit/assets/6229031/a19bd822-55a1-4b4d-af80-92f130c6e715)

I chose this design initially, so we stay compatible with the signature of the distance method inside pyRiemann.

Reply:
OK. These global optimizer will be resolved next time on a live meeting for example.